lib/Parse/ParseIfConfig.cpp

```diff
@@ -623,8 +623,18 @@ ParserResult<IfConfigDecl> Parser::parseIfConfig(
     foundActive |= isActive;
 
     if (!Tok.isAtStartOfLine() && Tok.isNot(tok::eof)) {
-      diagnose(Tok.getLoc(),
-               diag::extra_tokens_conditional_compilation_directive);
+      if (isElse && Tok.is(tok::kw_if)) {
+        diagnose(Tok.getLoc(),
+                 diag::unexpected_if_following_else_compilation_directive)
+            .fixItReplace(
+                SourceRange(ClauseLoc, Tok.getLoc().getAdvancedLoc(1)),
```

Comment:
`.getAdvancedLoc(1)` shouldn't be needed.

Reply:
fixed